fem/nonlinearform.hpp

```diff
@@ -22,6 +24,13 @@ namespace mfem
 class NonlinearForm : public Operator
 {
 protected:
+   /// The form assembly level (full, partial, etc.)
+   AssemblyLevel assembly;
```

Comment:
Full and element assembly levels do not seem to make sense for the action of a nonlinear operator.

On the other hand, they do make sense for the gradient. Generally, it is possible to have independent assembly levels for the nonlinear operator and its gradient.

Reply:
Do you mean changing the comment is enough?

Reply:
Yes, the comment here, the comment for the `ext` data member, and the comment for `SetAssemblyLevel`. Also the default behavior is not `AssemblyLevel::FULL`, it is a form of `AssemblyLevel::NONE`. The code in `SetAssemblyLevel` probably should be updated too.

Reply:
I took a shot at it. Let me know if you want me to change anything.